src/wallet/rpcwallet.cpp

```diff
@@ -2298,6 +2298,14 @@ UniValue walletpassphrase(const JSONRPCRequest& request)
     // Alternately, find a way to make request.params[0] mlock()'d to begin with.
     strWalletPass = request.params[0].get_str().c_str();
 
+    // Get the timeout
+    int nSleepTime = request.params[1].get_int();
+    // Timeout cannot be negative, otherwise it will relock immediately
+    if (nSleepTime < 0) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Timeout cannot be negative.");
+    }
+    pwallet->nRelockTime = GetTime() + nSleepTime;
```

Comment:
Still need to check the passphrase etc first... otherwise we've essentially tricked the wallet into lengthening its unlock time.

Reply:
Displaying the wrong time is kind of a problem, although maybe not a security one.